main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification.scala

```diff
@@ -124,6 +124,12 @@ object EffUnification {
         case (_, Type.Cst(TypeConstructor.Error(_), _)) =>
           return Ok((Substitution.empty, Nil))
 
+        case (t1@Type.Var(x, _), t2) if renv0.isFlexible(x) && !t2.typeVars.contains(t1) =>
```

Comment:
I would like this change to EffUnification to be in one small PR, to gauge performance cost.

Reply:
And correctness under the old system.

Reply:
This breaks in the old system: https://github.com/flix/flix/pull/7275

Reply:
That's fucked up. But I see the real fault is in our region stuff.

Reply:
Ok, alright. We leave it here.